yarn-project/noir-contracts/src/contracts/inclusion_proofs_contract/src/main.nr

```diff
@@ -0,0 +1,230 @@
+        },
+        context::Context,
+        note::{
+            note_getter_options::NoteGetterOptions,
+            note_header::NoteHeader,
+            utils as note_utils,
+        },
+        constants_gen::{
+            NOTE_HASH_TREE_HEIGHT,
+            HISTORIC_BLOCKS_TREE_HEIGHT,
+            PUBLIC_DATA_TREE_HEIGHT,
+            GENERATOR_INDEX__PUBLIC_LEAF_INDEX,
+        },
+        oracle::{
+            get_block_data::get_block_data,
+            get_membership_witness::{
+                get_membership_witness,
+                MembershipWitness,
+            },
+            get_sibling_path::get_sibling_path,
+            get_low_nullifier_membership_witness::{
+                get_low_nullifier_membership_witness,
+                LowNullifierMembershipWitness,
+            },
+        },
+        hash::pedersen_hash,
+    };
+    use dep::value_note::value_note::{ValueNote, ValueNoteMethods, VALUE_NOTE_LEN};
+
+    use crate::utils::{
+        full_field_less_than,
+        full_field_greater_than,
+    };
+
+    struct Storage {
+        private_values: Map<Set<ValueNote, VALUE_NOTE_LEN>>,
+        public_value: PublicState<Field, 1>,
+    }
+
+    impl Storage {
+        fn init(context: Context) -> Self {
+            Storage {
+                private_values: Map::new(
+                    context,
+                    1, // Storage slot
+                    |context, slot| {
+                        Set::new(context, slot, ValueNoteMethods)
+                    },
+                ),
+                public_value: PublicState::new(
+                    context,
+                    2, // Storage slot
+                    FieldSerializationMethods,
+                ),
+            }
+        }
+    }
+
+    #[aztec(private)]
+    fn constructor(public_value: Field) {
+        let selector = compute_selector("_initialize(Field)");
+        context.call_public_function(context.this_address(), selector, [public_value]);
+    }
+
+    #[aztec(public)]
+    internal fn _initialize(value: Field) {
+        storage.public_value.write(value);
+    }
+
+    // Creates a value note owned by `owner`.
+    #[aztec(private)]
+    fn create_note(
+        owner: AztecAddress,
+        value: Field,
+    ) {
+        let owner_private_values = storage.private_values.at(owner.address);
+        let mut note = ValueNote::new(value, owner.address);
+        owner_private_values.insert(&mut note, true);
+    }
+
+
+    // Proves that the owner owned a ValueNote at block `block_number`.
+    #[aztec(private)]
+    fn proveNoteInclusion(
+        owner: AztecAddress,
+        block_number: Field, // The block at which we'll prove that the note exists
+    ) {
+        // TODO: assert that block number is less than the block number of context.block_data
+        // --> This will either require a new oracle method that returns block_data.global_variables_hash preimage
```

Comment:
I would prefer the oracle for one main reason: 
If we expose it in the context, I believe people will think that it is the current one because that is kinda how you would use it in public etc.

Reply:
But having the oracle would result in more hashing because I would have to hash there the preimage to check that it matches the globals hash. Don't you think having a function `PrivateContext::block_data_block_number` or something like `HistoricBlockData::data_block_number` would be enough?

Reply:
Maybe the `HistoricBlockData` one, but generally don't want to make it too easy for people to blow off their own heads by being stupid.

Reply:
What about `PrivateContext::historic_block_data_block_number`